### ISSUE TITLE
`-fmax-errors` not support by clang

### ISSUE THREAD

**Issue by pbrady** [lanl/singularity-eos#315]
https://github.com/lanl/singularity-eos/blob/main/CMakeLists.txt#L449 unconditionally adds `-fmax-errors=3` which leads to the following warning:

```
clang: warning: argument unused during compilation: '-fmax-errors=3' [-Wunused-command-line-argument]
```

**Comment by Yurlungur** [comment]
Good catch---we should add a clang build to our CI... In the meantime, @mauneyc-LANL, @rbberger and/or @dholladay00 maybe we add that in with a generator expression for non-clang builds only?

**Comment by mauneyc-LANL** [comment]
This is my fault. Usually when developing I put this in so that I have a reasonable shot of locating the compiler error before my terminal burns through main memory printing out 1000s of incomplete template noise.  

We should really just remove it altogether; any objections? @Yurlungur @dholladay00 @rbberger

**Comment by Yurlungur** [comment]
None---let's just remove it. Thanks, @mauneyc-LANL

**Comment by mauneyc-LANL** [comment]
> Good catch---we should add a clang build to our CI... In the meantime, @mauneyc-LANL, @rbberger and/or @dholladay00 maybe we add that in with a generator expression for non-clang builds only?

We should use the intel compilers in our CI if we arn't already, too

**Comment by Yurlungur** [comment]
Yeah @dholladay00 brought that up recently. I think he's working on making that happen.

**Comment by dholladay00** [comment]
> Yeah @dholladay00 brought that up recently. I think he's working on making that happen.

I am fixing some issues with intel builds, but I haven't put in an intel build in our CI. I think that would require additional spacks buildouts and interacting with the project space in ways I am not very familiar with. Perhaps @ktsai7 or @rbberger could look into adding such a thing to our CI or replacing a current gnu build with an intel.

**Comment by ktsai7** [comment]
@dholladay00 sure, we can add an intel CI job, but do we want a new CI job or replace an existing one? Which intel compiler?

**Comment by dholladay00** [comment]
@ktsai7 I'm looking and there are several gnu Skylake tests. I'd say replace `test_gnu_skylake` with an intel test that turns on all of the bells and whistles (fortran, closure, kokkos and kernels, eospac, spiner), but I could be convinced to just add this in addition.

**Comment by ktsai7** [comment]
I'm fine with either way, but I do need to know which intel compiler do you want to use?

**Comment by mauneyc-LANL** [comment]
@ktsai7 @dholladay00 we should be using the oneAPI compilers (classic intel C/C++ compilers are losing support), but there's still broad downstream use of classic compilers.

The oneAPI compilers use a clang frontend, so having a compile with just a `clang++` would be good to be out in front of the eventual changeover.

**Comment by Yurlungur** [comment]
We now build with `clang++` on the github side and we build with the `cce` in our re-git build matrix. Still no explicit intel compiler build though.

**Comment by rbberger** [comment]
@Yurlungur that's now easy to add. Remind me in the new year.
